src/components/ContactsList/ContactRow.jsx

```diff
@@ -20,35 +21,24 @@ class ContactRow extends Component {
       onClick,
       breakpoints: { isMobile }
     } = this.props
-    const { number: phone } = getPrimaryOrFirst(contact.phone) || {
-      number: undefined
-    }
-
-    const { address: email } = getPrimaryOrFirst(contact.email) || {
-      address: undefined
-    }
-    const name = contact.name || {}
-    const isMyself = contact.metadata ? !!contact.metadata.me : false
-    const cozyUrl = getPrimaryOrFirst(contact.cozy) || { url: undefined }
+    const email = getPrimaryEmail(contact) || undefined
+    const phone = getPrimaryPhone(contact) || undefined
+    const cozyUrl = getPrimaryCozy(contact) || undefined
+
     return (
       <div className="contact" onClick={onClick}>
         <ContactWithSelection contact={contact} />
-        <ContactIdentity name={name} myself={isMyself} />
+        <ContactIdentity contact={contact} />
         {!isMobile && <ContactEmail email={email} />}
         {!isMobile && <ContactPhone phone={phone} />}
-        {!isMobile && <ContactCozy cozyUrl={cozyUrl.url} />}
+        {!isMobile && <ContactCozy cozyUrl={cozyUrl} />}
       </div>
     )
   }
 }
-ContactRow.propTypes = {
-  contact: PropTypes.shape({
-    name: contactPropTypes.name,
-    email: PropTypes.arrayOf(contactPropTypes.email),
-    phone: PropTypes.arrayOf(contactPropTypes.phone),
-    metadata: contactPropTypes.metadata
-  }).isRequired,
 
+ContactRow.propTypes = {
+  contact: PropTypes.object.isRequired,
```

Comment:
why did you remove the shape?

Reply:
Because used like this on other places, so I picked the wrong example.
Now they are all fixed and using shape